src/algorithms.jl

```diff
@@ -76,10 +76,7 @@ Finally, the same behavior is obtained when the keyword arguments are
 passed as the third positional argument in the form of a `NamedTuple`. 
 """ select_algorithm
 
-function select_algorithm(f::F, A, alg::Alg=nothing; kwargs...) where {F,Alg}
-    return select_algorithm(f, typeof(A), alg; kwargs...)
-end
-function select_algorithm(f::F, ::Type{A}, alg::Alg=nothing; kwargs...) where {F,A,Alg}
+function select_algorithm(f::F, A::T, alg::Alg=nothing; kwargs...) where {F,T,Alg}
```

Comment:
Is the type parameter for `A` necessary here to force specialisation for when `A` is a `Type`?  Or can it be omitted?

Reply:
Good question, I can try to remove it.